src/main/java/org/broadinstitute/consent/http/db/mapper/DatasetReducer.java

```diff
@@ -40,12 +41,24 @@ public void accumulate(Map<Integer, Dataset> map, RowView rowView) {
         && hasColumn(rowView, "propertyvalue", String.class)) {
       String keyName = rowView.getColumn("key", String.class);
       String propVal = rowView.getColumn("propertyvalue", String.class);
+      String schemaProp = rowView.getColumn("schemaproperty", String.class);
+      DatasetPropertyType propType = DatasetPropertyType.String;
+      if (hasColumn(rowView, "propertytype", String.class)) {
+          propType = DatasetPropertyType.parse(rowView.getColumn("propertytype", String.class));
+      }
+
       if (Objects.nonNull(keyName) && Objects.nonNull(propVal)) {
-        DatasetProperty prop = new DatasetProperty();
-        prop.setDataSetId(dataset.getDataSetId());
-        prop.setPropertyValue(propVal);
-        prop.setPropertyName(keyName);
-        dataset.addProperty(prop);
+        try {
+          DatasetProperty prop = new DatasetProperty();
+          prop.setDataSetId(dataset.getDataSetId());
+          prop.setPropertyValue(propType.coerce(propVal));
+          prop.setPropertyName(keyName);
+          prop.setPropertyType(propType);
+          prop.setSchemaProperty(schemaProp);
+          dataset.addProperty(prop);
+        } catch (Exception e) {
+          // do nothing.
```

Comment:
is doing nothing appropriate here?

Reply:
Yes, this is OK for now. We have another ticket about handling these in a consistent fashion: https://broadworkbench.atlassian.net/browse/DUOS-1417